servers/rendering/renderer_rd/effects/tone_mapper.cpp

```diff
@@ -216,9 +217,11 @@ void ToneMapper::tonemapper(RD::DrawListID p_subpass_draw_list, RID p_source_col
 	RID default_sampler = material_storage->sampler_rd_get_default(RS::CANVAS_ITEM_TEXTURE_FILTER_LINEAR, RS::CANVAS_ITEM_TEXTURE_REPEAT_DISABLED);
 	RID default_mipmap_sampler = material_storage->sampler_rd_get_default(RS::CANVAS_ITEM_TEXTURE_FILTER_LINEAR_WITH_MIPMAPS, RS::CANVAS_ITEM_TEXTURE_REPEAT_DISABLED);
 
+	RID viewport_scale_sampler = material_storage->sampler_rd_get_default(p_filter_mode, RS::CANVAS_ITEM_TEXTURE_REPEAT_DISABLED);
 	RD::Uniform u_source_color;
 	u_source_color.uniform_type = RD::UNIFORM_TYPE_INPUT_ATTACHMENT;
 	u_source_color.binding = 0;
+	u_source_color.append_id(viewport_scale_sampler);
```

Comment:
I'm a bit confused about this, you are passing in a sampler, but its in an input attachment, not a `UNIFORM_TYPE_SAMPLER_WITH_TEXTURE`. Additionally, clearly this was working previously without a linear sampler, and yet there are no shader changes to accommodate  the sampler. 

How does this work?